test/integration/helpers.sh

```diff
@@ -267,12 +267,13 @@ function start_up() {
 
         echo "Running tpm2_startup -c"
         tpm2_startup -c
+    elif [ -n "$TPM2_DEVICE" ]; then
```

Comment:
I would just mark TPM2TOOLS_TCTI in the Makefile.am as exported in AM_TESTS_ENVIRONMENT and then in the script if TPM2TOOLS_TCTI is empty, default to abrmd. This way at Make someone can  just point to another TCTI if they so choose.

Reply:
In my opinion TPM2TOOLS_TCTI should not be able to set directly during make process, only indirectly with TPM2_ABRMD, TPM2_SIM and TPM2_DEVICE. (service abrmd is missing!).
This allows maximum flexibility, for example if you want to reuse the test environment (helpers.sh) to write your own scripts.
Finally I am not sure, what is the best solution in case of neither TPM2_ABRMD, TPM2_SIM, TPM2_DEVICE nor TPM2TOOLS_TCTI is set:
1. error
2. set something specific, e.g. "device" or "abrmd" (not my favorite)
3. use find_default_tcti logic (see tpm2_options.c)

Reply:
Doesn't this work:
  - https://www.gnu.org/software/automake/manual/html_node/Scripts_002dbased-Testsuites.html

The AM_TESTS_ENVIRONMENT and TESTS_ENVIRONMENT variables can be used to run initialization code and set environment variables for the test scripts. The former variable is developer-reserved, and can be defined in the Makefile.am, while the latter is reserved for the user, which can employ it to extend or override the settings in the former; for this to work portably, however, the contents of a non-empty AM_TESTS_ENVIRONMENT must be terminated by a semicolon.

Reply:
> In my opinion TPM2TOOLS_TCTI should not be able to set directly during make process, only indirectly with TPM2_ABRMD, TPM2_SIM and TPM2_DEVICE. (service abrmd is missing!).
> This allows maximum flexibility, for example if you want to reuse the test environment (helpers.sh) to write your own scripts.

Running the test harness outside of make check, while feasible is not going to be the recommended or supported approach. Passing env variables through automake is the proper way, and if you're doing it out of the make file environment, its on the caller to set it up.

Reply:
Also their is no need to have different env variables then what the tools support already.